tls/extensions/s2n_server_key_share.c

```diff
@@ -43,15 +43,15 @@ static int s2n_server_key_share_send(struct s2n_connection *conn, struct s2n_stu
 
     /* Retry requests only require the selected named group, not an actual share.
      * https://tools.ietf.org/html/rfc8446#section-4.2.8 */
-    if (s2n_is_hello_retry_required(conn)) {
+    if (s2n_is_hello_retry_message(conn)) {
         notnull_check(conn->secure.server_ecc_evp_params.negotiated_curve);
 
         /* There was a mutually supported group, so that is the group we will select */
         uint16_t curve = conn->secure.server_ecc_evp_params.negotiated_curve->iana_id;
         GUARD(s2n_stuffer_write_uint16(out, curve));
         return 0;
     }
-
+    conn->secure.server_ecc_evp_params.evp_pkey = NULL;
```

Comment:
Why is this necessary?

Reply:
In the next line, a call to `s2n_ecdhe_parameters_send` expects `conn->secure.server_ecc_evp_params.evp_pkey` to be NULL otherwise it errors. We are making sure there is no garbage value here. 

I could add a check prior to `s2n_server_key_share_send` instead and set the value to NULL prior to ``s2n_server_key_share_send`` being called, to catch this error earlier.